deploy/wip-catalog/templates/apiserver.yaml

```diff
@@ -37,11 +37,18 @@ metadata:
   name: apiserver
 spec:
 {{ if .Values.debug }}
+{{ if .Values.nodePort }}
+  type: NodePort
+{{ else }}
   type: LoadBalancer
 {{ end }}
+{{ end }}
```

Comment:
A few things on this one. Bear with me.

Don't quite agree with the above few lines. It's not consistent with what I have seen on most other charts that give the option to specify what _type_ of service one wants.

Something like the following is more conventional:

https://github.com/kubernetes/charts/blob/master/stable/wordpress/templates/svc.yaml#L11

Going beyond the example I linked to, you could also default it right here (in line) using the `default` function. Alternatively, you can provide the default value in a `values.yaml` file right here in this same directory. Helm will automatically use that `values.yaml` and user specified settings or user-specific `values.yaml` just _overrides_ those defaults.

Lastly, services of `type: LoadBalancer` share all the characteristic of a service of `type: NodePort`. i.e. You get all the `NodePort` behavior for free on a `LoadBalancer`. Locally (e.g. on minikube) or on bare metal where the cloud provider no-ops load balancer provisioning, the external IP will simply remain in a pending state indefinitely, which is NBD. In other words-- you don't actually _need_ to expose this option in order to support local clusters. There _are_ defensible reasons that charts do (in general) tend to give an option re: service type, but it's got nothing to do with making them work locally. (Has more to do with an operator's chosen ingress strategy. They may or may not _want_ external load balancers.)